core/arch/arm/kernel/thread_spmc_a64.S

```diff
@@ -48,8 +48,13 @@ LOCAL_FUNC ffa_msg_send_direct_resp , :
 	mov	x2, #FFA_PARAM_MBZ			/* RES MBZ */
 
 .ffa_msg_loop:
-	/* Invoke SMC with caller provided parameters */
+	/*
+	 * Native interrupts unmasked while invoking SMC with caller
+	 * provided parameters.
+	 */
+	msr	daifclr, #DAIFBIT_NATIVE_INTR
```

Comment:
I gather the change is merged now. Just to state that I retested by reverting this change, and using upstream Hafnium which implements a proper fix https://review.trustedfirmware.org/c/hafnium/hafnium/+/25051
On OP-TEE  exiting the SPMC while a virtual interrupt is pending, the SPMC resumes again OP-TEE using the FFA_INTERRUPT interface signaling the virtual interrupt pending.

Reply:
Thanks, good to know.